test/subscription_map.hpp

```diff
@@ -258,54 +302,72 @@ class subscription_map_base {
 
     // Find all topic filters and allow modification
     template<typename Output>
-    void modify_match(MQTT_NS::string_view topic, Output callback) {
+    void modify_match(MQTT_NS::string_view topic, Output&& callback) {
         find_match_impl(*this, topic, std::forward<Output>(callback));
     }
 
-    template<typename Output>
-    void handle_to_iterators(handle h, Output output) {
+    template<typename ThisType, typename Output>
+    static void handle_to_iterators(ThisType& self, handle const &h, Output&& output) {
```

Comment:
Why did you make this a static function that takes a parameter to itself? Kind of defeats the point of member functions.

Reply:
To allow be called from const and non-const context. @redboltz applied the same to find_match

Reply:
Alright.